pkg/execution/runner/runner.go

```diff
@@ -534,6 +543,22 @@ func (s *svc) pauses(ctx context.Context, evt event.Event) error {
 }
 
 func (s *svc) initialize(ctx context.Context, fn inngest.Function, evt event.Event) error {
+	// Attempt to rate-limit the incoming function.
+	if s.rl != nil && fn.RateLimit != nil {
+		key, err := ratelimit.RateLimitKey(ctx, fn.ID, *fn.RateLimit, evt.Map())
+		if err != nil {
+			return err
+		}
+		limited, err := s.rl.RateLimit(ctx, key, *fn.RateLimit)
+		if err != nil {
+			return err
+		}
+		if limited {
+			// Do nothing.
+			return nil
```

Comment:
probably should still return an error, bubbling up to the user that execution has been rate limited.
people will think it's a bug when there's no reaction even thought it's the expected outcome for us.

and ignore that error in logging.

honestly it's best to be a warning instead of an error, but seems like it needs more changes to do that with the current structure.

Reply:
Yeah, I think we should probably tackle this UI side here and add something that shows rate limiting directly into the dev server UI.  It's a great shout, and something we can loop Ed into.